tests/unit/test_pipeline_graph.py

```diff
@@ -421,20 +421,111 @@ def test_merge_combined_implementations_iteration(default_config_params, test_di
     check_nodes_and_edges(pipeline_graph, expected_nodes, expected_edges)
 
 
-# TODO MIC-5466: Deduplicate slots so this fails with cycle error
-@pytest.mark.skip(reason="Not implemented")
-def test_cycle_error(default_config_params) -> None:
+def test_merge_combined_implementations_parallel(default_config_params, test_dir) -> None:
```

Comment:
This test doesn't seem to stress situations w/ duplicate names (only duplicate env var names). Not sure if that's important or actually possible or not.

Reply:
I think youd want to pass in a custom pipeline schema to do this properly. I added a test of _get_duplicate_slots() that ensures we will at least pick up duplicate names as being duplicated slots, though